### ISSUE TITLE
junit-xml does not output generated tests parameters

### ISSUE THREAD

**Issue by slemesle** [nose-devs/nose2#265]
I use generated tests with meaningfull parameters and I wish I could see them exported in junit-xml report. 

The patch looks pretty simple (junitxml.py): 

```python
    def testOutcome(self, event):
        """Add test outcome to xml tree"""
        test = event.test
        testid_lines = test.id().split('\n')
        testid = testid_lines[0]
        # split into module, class, method parts... somehow
        parts = testid.split('.')
        classname = '.'.join(parts[:-1])
        method = parts[-1]
        if len(testid_lines) > 1:
            method = '%s:' % method + ':'.join(testid_lines[1:])
```

With this one we can have all parameters as requested.

**Comment by slemesle** [comment]
Just created a Gist for that 

https://gist.github.com/slemesle/5d5c8e96d1d8a00e6aee

**Comment by little-dude** [comment]
It seems to be on purpose. In the unit tests, we have this test:
```python
def test_gen(self):
    def check(a, b):
        self.assertEqual(a, b)

    yield check, 1, 1
    yield check, 1, 2
```

and this check

```python
self.assertEqual(xml[0].get('name'), 'test_gen:1')
self.assertEqual(xml[1].get('name'), 'test_gen:2')
```

I guess the reason is sometimes, the arguments are not really printable or too long, so I'd rather not change it. Howver we could add an option to the `junit-xml` for this? Would that be ok @slemesle ?

**Comment by slemesle** [comment]
yes of course an option would be ok for me.

**Comment by little-dude** [comment]
I added a `test_fullname` option to the plugin (`False` by default): https://github.com/nose-devs/nose2/pull/269

```python
    # a.py
    from nose2 import tools

    def test_gen():
        def check(a, b):
            assert a == b, '{}!={}'.format(a,b)
        yield check, 99, 99
        yield check, -1, -1

    @tools.params('foo', 'bar')
    def test_params(arg):
        assert arg in ['foo', 'bar', 'baz']
```

Produces this XML by default:

```xml
    <testcase classname="a" name="test_gen:1" time="0.000171">
        <system-out />
    </testcase>
    <testcase classname="a" name="test_gen:2" time="0.000202">
        <system-out />
    </testcase>
    <testcase classname="a" name="test_params:1" time="0.000159">
        <system-out />
    </testcase>
    <testcase classname="a" name="test_params:2" time="0.000163">
        <system-out />
    </testcase>
```

With `test_fullname` set to `True`, the following XML is produced:

```xml
    <testcase classname="a" name="test_gen:1 (99, 99)" time="0.000213">
        <system-out />
    </testcase>
    <testcase classname="a" name="test_gen:2 (-1, -1)" time="0.000194">
        <system-out />
    </testcase>
    <testcase classname="a" name="test_params:1 ('foo')" time="0.000178">
        <system-out />
    </testcase>
    <testcase classname="a" name="test_params:2 ('bar')" time="0.000187">
        <system-out />
    </testcase>
```
would that be good?

**Comment by slemesle** [comment]
This looks perfect to me :) thanks a lot.

**Comment by little-dude** [comment]
this will be part of the next release.

**Comment by baggrey1** [comment]
Will the test_fullname option be included in one of the 0.6.x releases, or is it slated for 0.7?  I'm very much looking forward to having this feature included in a PyPI release!
